lib/print.dl

```diff
@@ -0,0 +1,12 @@
+// These functions are useful for debugging.
+// print can be inserted in imperative code,
+// while printTrue can be inserted on the RHS of rules
+// e.g. O(x) :- I(v), printTrue(v), x = v + 1.
+
+#[has_side_effects]
+extern function print(msg: string): ()
+
+function printTrue(msg: string): bool {
+   print(msg);
+   true
+}
```

Comment:
```suggestion
extern function debug<'T>(value: 'T): ()
```
Debug prints shouldn't be considered effectful and a specialization of the function that returns is redundant since we have `Inspect` clauses

Reply:
`has_side_effects` tells the compiler not to cache the result of the function, so it's absolutely necessary here.

Reply:
> specialization of the function that returns is redundant since we have Inspect clauses

`printTrue` can be used without `Inspect` though.  I don't mind either way.

Reply:
That's why printing should be "cached", so to speak: if printing forces re-computation you're no longer seeing the true behavior of the dataflow, you're seeing an altered version simply because our added a debug print

Reply:
All this attribute does is prevent converting function calls into `lazy_static`'s. It doesn't affect the dataflow in any way, but without it the output will only be produced the first time a rule containing the `print()` call is triggered.